services/supervisor/tests/supervisor/migrations/mod.rs

```diff
@@ -0,0 +1,879 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use exonum::{
+    crypto::Hash,
+    helpers::{Height, ValidatorId},
+    merkledb::access::Prefixed,
+    runtime::{
+        migrations::MigrationStatus, versioning::Version, CoreError, ErrorMatch, ExecutionError,
+        InstanceId, SUPERVISOR_INSTANCE_ID,
+    },
+};
+use exonum_rust_runtime::{DefaultInstance, ServiceFactory};
+use exonum_testkit::{ApiKind, TestKit, TestKitApi, TestKitBuilder};
+
+use exonum_supervisor::{
+    AsyncEventState, ConfigPropose, MigrationError, MigrationInfoQuery, MigrationRequest,
+    MigrationResult, MigrationState, SchemaImpl, Supervisor, SupervisorInterface,
+};
+
+use std::{thread, time::Duration};
+
+use crate::service_lifecycle::execute_transaction;
+
+use migration_service::{
+    FailingMigrationServiceV07, MigrationService, MigrationServiceV01_1, MigrationServiceV02,
+    MigrationServiceV05, MigrationServiceV05_1,
+};
+
+mod migration_service;
+
+/// Creates testkit with supervisor and three versions of migrating service.
```

Comment:
Nit: There are more than three versions of the service now, aren't there? I'd avoid using specific numbers for future compatibility 🙃